yt_dlp/extractor/rcs.py

```diff
@@ -349,10 +308,10 @@ class RCSIE(RCSBaseIE):
                         |corrierefiorentino\.
                     )?corriere\.it
                     |(?:gazzanet\.)?gazzetta\.it)
-                    /(?!video-embed/).+?/(?P<id>[^/\?]+)(?=\?|/$|$)'''
+                    /(?!video-embed/)\S+?/(?P<id>[^/\?]+)(?=\?|/$|$)'''
```

Comment:
```suggestion
                    /(?!video-embed/)[^?#]+?/(?P<id>[^/\?]+)(?=\?|/$|$)'''
```
URLs can technically have space. `.+?` is fine when matching against URL. The problem is when matching against large data (i.e, webpage)